ibis/expr/api.py

```diff
@@ -318,25 +318,96 @@ def schema(
         return sch.schema(names, types)
 
 
+_Row = TypeVar("_Row", Sequence, Mapping[str, Any])
+
+
 def table(
-    schema: SupportsSchema,
+    schema: SupportsSchema | None = None,
     name: str | None = None,
+    data: Sequence[_Row] | None = None,
+    columns: Iterable[str] | None = None,
 ) -> ir.Table:
-    """Create an unbound table for building expressions without data.
+    """Create a table literal or an abstract table without data.
 
     Parameters
     ----------
     schema
         A schema for the table
     name
         Name for the table. One is generated if this value is `None`.
+    data
+        Table literal data.
+    columns
+        A sequence of [`str`][str] column names.
 
     Returns
     -------
     Table
-        An unbound table expression
+        A table expression
+
+    Examples
+    --------
+    Create a table with no data backing it
+
+    >>> t = ibis.table(schema=dict(a="int", b="string"))
+    >>> t
+    UnboundTable: unbound_table_0
+      a int64
+      b string
```

Comment:
Since some users are already confused by unbound tables, perhaps we shouldn't highlight this calling pattern in an example? Up to you.

Reply:
Got rid of these.